### PR TITLE
Bugs/table style tooltip webapp path

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
#564 webapp path in config/env
#557 cell style
#558 cell tooltip

```
import pandas as pd

from taipy.gui import Gui, State

df = pd.DataFrame.from_records(
    [["yellow", "green", "tooltip for yellow"], ["red", "blue", "tooltip for red"]], columns=["A", "B", "C"]
)

def cellstyle(
    state: State, value: str, index: str, row: str, colname: str
) -> str:
    print("colname ", colname, " value ", value)
    return value + "-cell"

def rowstyle(
    state: State, index: str, row: str
) -> str:
    print("index ", index, " row ", row)
    return f"row-{index}"

page = (
    "<|{df}|table|width=400px|show_all=True"
    + "|style=rowstyle"
    + "|style[A]=cellstyle"
    + "|style[B]=cellstyle"
    + "|tooltip=cellstyle"
    + "|>"
)

if __name__ == "__main__":
    Gui(page=page, css_file="styletst").run()

```

### CODE REVIEW

**Review by dinhlongviolin1 — APPROVED** [review]
Works like a charm!

File: gui/doc/table.md [hunk 1]
```diff
@@ -135,3 +136,33 @@ Css definition
 }
 ```
 
+### Cell Tooltip
+
+You can specify a tooltip for specific table cells.
+
+When Taipy creates the cells, it can add a specific tooltip that you would have set as the
+return value of the function set to the `tooltip` or `tooltip[_column_name_]` property .
```
Comment (FabienLelaquais) [inline gui/doc/table.md]: set to the \_tooltip\_ or \_tooltip[column_name]\_ property.

File: gui/doc/table.md [hunk 2]
```diff
@@ -135,3 +136,33 @@ Css definition
...
+
+The signature of this function expects five optional parameters:
+     - _state_: the current state
+     - _value_: the value of the cell
+     - _index_: the index of the row in this table
+     - _row_: all the values for this row
+     - _column_name_: the name of the column for this cell
+
+Based on these parameters, the function must return a string that defines a tooltip that will
```
Comment (FabienLelaquais) [inline gui/doc/table.md]: "...be added to the cell." -> "... that is used as the cell's tooltip text."

File: gui/doc/viselements.json [hunk 3]
```diff
@@ -899,6 +899,16 @@
             "type": "str",
             "doc": "Allows the styling of table cells.\nSee <a href=\"#styling\">below</a> for details."
           },
+          {
+            "name": "tooltip",
+            "type": "str",
+            "doc": "The name of the function that will calculate the cell tooltips.\nSee <a href=\"#cell%20tooltip\">below</a> for details."
```
Comment (FabienLelaquais) [inline gui/doc/viselements.json]: "that will calculate the cell tooltips" -> "that must return a tooltip text for a cell"

File: gui/doc/viselements.json [hunk 4]
```diff
@@ -899,6 +899,16 @@
...
+          },
+          {
+            "name": "tooltip[<i>col_name</i>]",
+            "type": "str",
+            "doc": "The name of the function that will calculate the cell tooltips.\nSee <a href=\"#cell%20tooltip\">below</a> for details."
```
Comment (FabienLelaquais) [inline gui/doc/viselements.json]: (same change)

File: src/taipy/gui/_gui_cli.py [hunk 5]
```diff
@@ -23,6 +23,7 @@ def _create_parser(cls):
         gui_parser.add_argument("-H", "--host", nargs="?", default="", const="", help="Specify server host")
 
         gui_parser.add_argument("--ngrok-token", nargs="?", default="", const="", help="Specify NGROK Authtoken")
+        gui_parser.add_argument("--webapp-path", nargs="?", default="", const="", help="Specify webapp path")
```
Comment (FabienLelaquais) [inline src/taipy/gui/_gui_cli.py]: "Specify webapp path" -> "The path to the WebApp to be used. The default is the webapp directory under gui in the Taipy GUI package directory."

File: src/taipy/gui/data/pandas_data_accessor.py [hunk 6]
```diff
@@ -74,13 +74,25 @@ def __build_transferred_cols(
...
+                    col_applied = self.__apply_user_function(gui, func, k if k in cols else None, v, data, "tps__")
+                if not col_applied:
                     data[v] = v
-                cols.append(v)
+                cols.append(col_applied or v)
+        if tooltips:
+            # copy the df so that we don't "mess" with the user's data
+            if not is_copied:
+                # copy the df so that we don't "mess" with the user's data
+                data = data.copy()
```
Comment (FabienLelaquais) [inline src/taipy/gui/data/pandas_data_accessor.py:88]: No performance issue I guess? Just copying references?

File: src/taipy/gui/data/pandas_data_accessor.py [hunk 7]
```diff
@@ -109,18 +121,18 @@ def __build_transferred_cols(
...
+                _PandasDataAccessor.__user_function,
                 axis=1,
                 args=(gui, column_name, user_function, function_name),
             )
-            return True
-        else:
-            warnings.warn(f"Style function {function_name} should take at least a value as second argument")
+            return new_col_name
+        except Exception as e:
+            warnings.warn(f"Exception raised when applying user function {function_name}.\n{e}")
```
Comment (FabienLelaquais) [inline src/taipy/gui/data/pandas_data_accessor.py]: -> ... when invoking user function...?

**Review by FabienLelaquais — APPROVED** [review]
Minor user-facing comments...
Great job!

File: src/taipy/gui/data/pandas_data_accessor.py (hunk 6 above)
Comment (FredLL-Avaiga) [reply src/taipy/gui/data/pandas_data_accessor.py:88]: not copying reference but we never modify the user's data

**Pull request merged**